### PR TITLE
(PA-315) Add el-6-s390x platform

### PR DESCRIPTION
**Pull request by ScottGarman**
This platform is for RHEL running on IBM z-series

### CODE REVIEW

**Comment by ScottGarman** [comment]
Running this through a jenkins ad-hoc build to ensure no breakage for the non cross compiled EL platforms: https://jenkins.puppetlabs.com/view/puppet-agent/view/ad%20hoc/job/platform_puppet-agent_init-van-int_suite-manual-ad-hoc/144/

**Comment by mckern** [comment]
> This platform is for RHEL running on **IBM z-series**.

File: configs/components/ruby-selinux.rb [hunk 1]
```diff
@@ -20,10 +19,16 @@
...
+  if platform.is_cross_compiled_linux?
+    pkg.environment "RUBY" => settings[:host_ruby]
+    ruby = "#{settings[:host_ruby]} -r#{settings[:datadir]}/doc/rbconfig.rb"
+  end
+
   pkg.build do
-    ["export RUBYHDRDIR=$(shell #{settings[:bindir]}/ruby -rrbconfig -e 'puts RbConfig::CONFIG[\"rubyhdrdir\"]')",
-     "export VENDORARCHDIR=$(shell #{settings[:bindir]}/ruby -rrbconfig -e 'puts RbConfig::CONFIG[\"vendorarchdir\"]')",
-     "export ARCHDIR=$${RUBYHDRDIR}/$(shell #{settings[:bindir]}/ruby -rrbconfig -e 'puts RbConfig::CONFIG[\"arch\"]')",
+    ["export RUBYHDRDIR=$(shell #{ruby} -e 'puts RbConfig::CONFIG[\"rubyhdrdir\"]')",
```
Comment (mckern) [inline configs/components/ruby-selinux.rb]: This is more of a personal nitpick so it's nothing I'd block over. But when you've got multiple kinds of nested quotations going on like this, consider the humble `%()`:

```ruby
%(export RUBYHDRDIR=$(shell #{settings[:bindir]}/ruby -rrbconfig -e 'puts RbConfig::CONFIG["rubyhdrdir"]'))
```

A little less escaping, a slightly cleaner string (I think).

File: configs/components/openssl.rb [hunk 2]
```diff
@@ -11,8 +11,9 @@
     pkg.build_requires "pl-gcc-#{platform.architecture}"
     pkg.build_requires 'runtime' if platform.is_huaweios?
     # needed for the makedepend command
+    pkg.build_requires 'imake' if platform.name =~ /^el/
```
Comment (mckern) [inline configs/components/openssl.rb]: Have we never needed `imake` before?

Comment (ScottGarman) [inline configs/components/openssl.rb]: That's a good question. I think something about customizing the target arch for cross-compiles invokes makedepend in a way that native builds do not. But I haven't dug into that. 

Either way, without this the build clearly fails with an error about needing the makedepend command, and succeeds when it has it. Distros are all over the map in how they package it.

Comment (geoffnichols) [inline configs/components/openssl.rb]: Not sure if this is useful, but [this thread](https://bugzilla.redhat.com/show_bug.cgi?id=890712) might be tangentially related to the `imake` question.

**Comment by geoffnichols** [comment]
👍

**Pull request merged**